src/store.js

```diff
@@ -13,11 +13,20 @@ Vue.use(Vuex);
 const readerStore = createReaderStore();
 const skeletonStore = createSkeletonStore();
 
-export default new Vuex.Store({
+const store = new Vuex.Store({
   modules: {
     [readerStore.namespace]: readerStore.store,
     [scaifeWidgets.namespace]: scaifeWidgets.store,
     [skeletonStore.namespace]: skeletonStore.store,
   },
   ...createStore(),
 });
+
+if (process.env.VUE_APP_TOC_ENDPOINT) {
+  // TODO: Determine if there is a better way to make this
```

Comment:
@paltman: This is a pattern we're likely going to encounter a lot.

- We need to define an endpoint in `scaife-widgets`.  It might end up being a third-party API (like for the New Alexandria Commentary), or it might be a site-specific API
- Within a Vue project, I'd typically want to override using an environment variable and fallback to a default
- I don't think I can really do that with `scaife-widgets`, because of how the module is built and released on npm

My workaround had been to define the endpoint in the store within `scaife-widgets`:

https://github.com/scaife-viewer/scaife-widgets/pull/20/files#diff-e3b1ca4ad5a207c170c97a435b3d1ff3R10

```javascript
const state = {
  readerTextSize: 'md',
  readerTextWidth: 'normal',
  // TODO: Determine if there is a better way to override this
  tocEndpoint:
    process.env.VUE_APP_TOC_ENDPOINT ||
    'https://mini-stack-a-feature-se-j47yu0.herokuapp.com',
};
```

and then override it within the site repo.

Have you run across this pattern in other projects / elsewhere in the Vue ecosystem?  Can you think of a better way to address this type of configuration within a project?

Reply:
(https://github.com/scaife-viewer/scaife-widgets/pull/19 is another concrete where we've defined endpoints in the widgets that we may need to override within a project)